### PR TITLE
[Jira] Handle empty response when status code is 200

### PR DESCRIPTION
**Pull request by moxarth-elastic**
## Part of https://github.com/elastic/connectors/issues/2480

Raise EmptyResponseError if the response body is empty when the status code is 200.

#### Pre-Review Checklist
- [x] this PR does NOT contain credentials of any kind, such as API keys or username/passwords (double check `config.yml.example`)
- [x] this PR has a meaningful title
- [x] this PR links to all relevant github issues that it fixes or partially addresses
- [x] if there is no GH issue, please create it. Each PR should have a link to an issue
- [x] this PR has a thorough description
- [x] Covered the changes with automated tests
- [x] Tested the changes locally
- [x] Added a label for each target release version (example: `v7.13.2`, `v7.14.0`, `v8.0.0`)

### CODE REVIEW

**Comment by moxarth-elastic** [comment]
buildkite test this

**Comment by moxarth-elastic** [comment]
buildkite test this

File: connectors/sources/jira.py [hunk 1]
```diff
@@ -242,6 +248,9 @@ async def api_call(self, url_name=None, **url_kwargs):
                     url=url,
                     ssl=self.ssl_ctx,
                 ) as response:
+                    if response.content_length == 0:
+                        msg = "The response body is empty"
```
Comment (seanstory) [inline connectors/sources/jira.py:252]: Let's include the URL in the error message, to facilitate troubleshooting.

**Review by seanstory — COMMENTED** [review]
minor nit

**Review by artem-shelkovnikov — APPROVED** [review]

**Comment by seanstory** [comment]
@moxarth-elastic it looks like this PR was not correctly labeled to ensure that the bugfix was backported to the most recent minor version. I've added the labels above, but I don't believe our backport automation will catch it at this point. Can you please manually trigger backports for the 8.13 and 8.14 branches?

**Comment by moxarth-elastic** [comment]
> @moxarth-elastic it looks like this PR was not correctly labeled to ensure that the bugfix was backported to the most recent minor version. I've added the labels above, but I don't believe our backport automation will catch it at this point. Can you please manually trigger backports for the 8.13 and 8.14 branches?

Sure, i'll do it

**Comment by moxarth-elastic** [comment]
@seanstory backport PRs to 8.13 and 8.14
- https://github.com/elastic/connectors/pull/2537
- https://github.com/elastic/connectors/pull/2536

**Pull request merged**